### DISCUSSION TITLE
send microphone-stream to Server

### DISCUSSION THREAD

**Discussion by reschresch** [naudio/NAudio#1189] [Q&A]
Hello,
I want to send an audio-stream from my microphone to a streaming-server so that a few (e.g. 5) people can listen what I speak into my microphone.
Recording from my mic is no problem but how I can convert this wave-stream into G.722? 
And have NAudio an (implemented) function to send this stream to my media-Server (e.g. in RTSP)?

It would be great if someone had an idea for me.
Best regards,
Rene

**Comment by markheath** [comment]
NAudio covers the capture side well - use `WasapiRecorderBuilder().Build()` (its `DataAvailable` gives you a zero-copy `ReadOnlySpan<byte>`, or use `CaptureAsync` with `await foreach`). 

But NAudio doesn't ship a G.722 encoder or any RTSP/RTP streaming stack - those are out of scope. 

For G.722 you'd need to bring your own codec, or you might want to consider using a library like Concentus for Opus - see the Network chat demo in the NAudio demo app in the /samples folder.

For transport you'd use something like a dedicated RTSP/WebRTC library.
